rotkehlchen/chain/evm/decoding/aave/common.py

```diff
@@ -171,36 +204,75 @@ def _decode_deposit(
                 ):
                     event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
                     resolved_asset = event.asset.resolve_to_asset_with_symbol()
-                    event.notes = f'Receive {event.balance.amount} {resolved_asset.symbol} from {self.label}'  # noqa: E501
+                    event.balance.amount = amount
+                    event.notes = f'Receive {amount} {resolved_asset.symbol} from {self.label}'
                     event.counterparty = self.counterparty
                     receive_event = event
 
-        return deposit_event, receive_event
+        return deposit_event, receive_event, earned_event
 
     def _decode_withdrawal(
             self,
             token: 'EvmToken',
-            tx_log: EvmTxReceiptLog,
-            decoded_events: list['EvmEvent'],
-    ) -> tuple[EvmEvent | None, EvmEvent | None]:
+            context: 'DecoderContext',
+    ) -> tuple[EvmEvent | None, EvmEvent | None, EvmEvent | None]:
         """Decode aave v2/v3 withdrawal event. Returns the Return and Withdraw events."""
-        user = hex_or_bytes_to_address(tx_log.topics[2])
-        to = hex_or_bytes_to_address(tx_log.topics[3])
+        reserve = hex_or_bytes_to_address(context.tx_log.topics[1])
+        user = hex_or_bytes_to_address(context.tx_log.topics[2])
+        to = hex_or_bytes_to_address(context.tx_log.topics[3])
         return_event, withdraw_event = None, None
         if (
             not (is_wnative_user := user in self.native_gateways) and
             not self.base.any_tracked([user, to])
         ):
-            return None, None
+            return None, None, None
         amount = asset_normalized_value(
-            amount=hex_or_bytes_to_int(tx_log.data),
+            amount=hex_or_bytes_to_int(context.tx_log.data),
             asset=token,
         )
+
+        earned_event = None  # find the mint or burn event of aToken
+        for _log in context.all_logs:
+            if (
+                _log.topics[0] in {MINT, BURN} and
+                user == hex_or_bytes_to_address(_log.topics[1]) and ((
+                    to is not None and
+                    hex_or_bytes_to_address(_log.topics[2]) == to
+                ) or to is None) and (
+                    balance_increase := asset_normalized_value(
+                        amount=hex_or_bytes_to_int(_log.data[32:64]),
+                        asset=token,
+                    )
+                ) > 0
+            ):  # parse the mint amount and balance_increase
+                earned_token_address = _log.address
+                if _log.topics[0] == BURN:
+                    amount = asset_normalized_value(
+                        amount=hex_or_bytes_to_int(_log.data[:32]),
+                        asset=token,
+                    )
+                    earned_token_address = reserve
```

Comment:
`accrued_token_address` here. I believe is clearer here becayuse you can accrue debt that you pay I believe